Objects/weakrefobject.c

```diff
@@ -46,38 +46,54 @@ new_weakref(PyObject *ob, PyObject *callback)
 }
 
 
-/* This function clears the passed-in reference and removes it from the
- * list of weak references for the referent.  This is the only code that
- * removes an item from the doubly-linked list of weak references for an
- * object; it is also responsible for clearing the callback slot.
+/* This function is responsible for clearing the callback slot. When using this
+ * It must be called before calling remove_weakref, otherwise a segmentation fault will 
+ * occur at build time.
```

Comment:
I believe the second sentence could make use of a slight rephrase.

Reply:
Thanks, I have reworded it now.